### ISSUE TITLE
RuntimeError: 3D or 4D (batch mode) tensor expected for input

### ISSUE THREAD

**Issue by MortenKoppel** [alumae/kiirkirjutaja#2]
Occasionally the running container throws the error (below) and halts. Sometimes it runs for tens of minutes without errors but sometimes under a minute.  Just for the reference the test machine had a fairly modest CPU: Intel(R) Core(TM) i3-7100. The container was started with the following command:

```
sudo docker run --shm-size=20gb \
 -it koodivaramu.eesti.ee:5050/taltechnlp/kiirkirjutaja python main.py \
 --youtube-caption-url http://upload.youtube.com/closedcaption?cid=(yourid) \
 rtmp://(yourstream)
```

This time it was started just after one crash and it crashed again quite soon, you can see the video and titles right before the crash here: https://youtu.be/1wuddvqdZJo?t=1800  and here is the last output from the container:

```
2021-10-21T07:38:23.097
 Ossinovski,
2021-10-21T07:38:23.637
 palun.

INFO:root:Response status 200 OK: 2021-10-21T07:38:24.226

Exception in thread Thread-22:
Traceback (most recent call last):
  File "/root/miniconda3/lib/python3.8/threading.py", line 932, in _bootstrap_inner
    self.run()
  File "/root/miniconda3/lib/python3.8/threading.py", line 870, in run
    self._target(*self._args, **self._kwargs)
  File "/opt/kiirkirjutaja/asr.py", line 33, in run
    for chunk in self.chunk_generator:
  File "/opt/kiirkirjutaja/lid.py", line 57, in filter
    language_id = self.get_language(buffer)
  File "/opt/kiirkirjutaja/lid.py", line 25, in get_language
    probs = self.get_language_probs(buffer)
  File "/opt/kiirkirjutaja/lid.py", line 19, in get_language_probs
    lang_logits, lang_group_logits = self.model(buffer.unsqueeze(0))
  File "/root/miniconda3/lib/python3.8/site-packages/torch/nn/modules/module.py", line 889, in _call_impl
    result = self.forward(*input, **kwargs)
RuntimeError: The following operation failed in the TorchScript interpreter.
Traceback of TorchScript, serialized code (most recent call last):
  File "code/__torch__/torch/nn/functional.py", line 61, in forward
          ops.prim.RaiseException(_5)
        if torch.eq(mode, "reflect"):
          _22 = torch.reflection_pad2d(input, pad)
                ~~~~~~~~~~~~~~~~~~~~~~ <--- HERE
          _21 = _22
        else:

Traceback of TorchScript, original code (most recent call last):
  File "/opt/conda/lib/python3.8/site-packages/torch/nn/functional.py", line 4014, in forward
            assert len(pad) == 4, "4D tensors expect 4 values for padding"
            if mode == "reflect":
                return torch._C._nn.reflection_pad2d(input, pad)
                       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ <--- HERE
            elif mode == "replicate":
                return torch._C._nn.replication_pad2d(input, pad)
RuntimeError: 3D or 4D (batch mode) tensor expected for input, but got: [ torch.FloatTensor{1,1,1,0} ]

Traceback (most recent call last):
  File "main.py", line 108, in <module>
    main(args)
  File "main.py", line 84, in main
    for res in turn_decoder.decode_results():
  File "/opt/kiirkirjutaja/asr.py", line 24, in decode_results
    result = self.result_queue.get()
  File "/root/miniconda3/lib/python3.8/queue.py", line 170, in get
    self.not_empty.wait()
  File "/root/miniconda3/lib/python3.8/threading.py", line 302, in wait
    waiter.acquire()
```

**Comment by liisaratsep** [comment]
@alumae - perhaps you can also update the `latest` tag to point to the fixed version?

**Comment by alumae** [comment]
Yes, my bad, "latest" was not updated. Looking for a good place to put the changelog.

**Comment by alumae** [comment]
We now have "releases", which are in sync with the Docker image. I.e., today we published release 0.2.3 and the Docker image is also updated (available as koodivaramu.eesti.ee:5050/taltechnlp/kiirkirjutaja:latest and koodivaramu.eesti.ee:5050/taltechnlp/kiirkirjutaja:0.2.3).
